FaceDetectionServer/REST-API.md

```diff
@@ -0,0 +1,240 @@
+# MobiledgeX Computer Vision Server REST API
+
+## Face Detection
+`/detector/detect/`
+
+Used to send a face image to the server and get back a set of coordinates for any detected faces.
+
+### POST
+`http://<hostname>:8008/detector/detect/`
+
+#### Parameters
+| Parameter | Description |
+| - | - |
+| image | The face(s) image. |
+
+#### Return
+JSON string with the following attributes.
+
+| Attribute | Description |
+| - | - |
+| success | "True" if face detected, otherwise "false". |
+| server_processing_time | The time in milliseconds that it took the server to process this image. |
+| rects | A JSON array of coordinates containing the detected faces. Only included if "success"="true". |
+
+#### Examples:
+##### Single face detected:
+```
+curl http://facedetection.defaultedge.mobiledgex.net:8008/detector/detect/ -F "image=@Bruce.jpg"
+{"success": "true", "server_processing_time": "19.873", "rects": [[73, 76, 147, 150]]}
+```
+
+##### Multiple faces detected:
+```
+curl http://facedetection.defaultedge.mobiledgex.net:8008/detector/detect/ -F "image=@face_triple.jpg"
+{"success": "true", "server_processing_time": "30.898", "rects": [[6, 104, 59, 157], [68, 165, 138, 235], [120, 28, 189, 97]]}
+```
+
+##### No faces detected:
+```
+curl http://facedetection.defaultedge.mobiledgex.net:8008/detector/detect/ -F "image=@single_pixel.png"
+{"success": "false", "server_processing_time": "0.030"}
+```
+
+## Face Recognition
+`/recognizer/predict/`
+
+Used to send a face image to the server and get back a set of coordinates for the recognized face, along with a "subject" name value if the face is recognized against the current trained data.
+
+_Note: This call is limited to a single face. At training time, no images are added to the set if more than one face is detected. If more than one face is present in an image for this call, prediction will only be performed on the first face detected._
+
+### POST
+```
+http://<hostname>:8008/recognizer/predict/
+```
+
+#### Parameters
+| Parameter | Description |
+| - | - |
+| image | The face image. |
+
+#### Return
+JSON string with a "success" value of "true" or "false", and if successful, the following JSON values:
+
+| Attribute | Description |
+| - | - |
+| success | "True" if face recognized, otherwise "false". |
+| subject | Name of detected face. |
+| confidence | Level of confidence. Actually a measure of distance the features are from matching a given subject. Lower is better, with 0 being a perfect match. |
+| rect | Coordinates of detected face. |
```

Comment:
"Coordinates of first detected face"

Reply:
also would be great to specify the meaning of the elements returned by the coordinates as mentioned above